src/libostree/ostree-repo-pull.c

```diff
@@ -2107,12 +2107,14 @@ ostree_repo_pull_with_options (OstreeRepo             *self,
         soup_uri_free (uri);
       }
 
-    if (bytes_sig && !_ostree_repo_load_cache_summary_if_same_sig (self,
-                                                                   remote_name_or_baseurl,
-                                                                   bytes_sig,
-                                                                   &bytes_summary,
-                                                                   cancellable,
-                                                                   error))
+    if (bytes_sig &&
+        !_ostree_repo_remote_name_is_file (remote_name_or_baseurl) &&
```

Comment:
Hm, can we call this function once and cache the result as `pull_data->is_local`?  Although looking again we already have `pull_data->remote_repo_local`.  Which should maybe be hoisted a lot earlier.